packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.tsx

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { useFtlMsgResolver } from '../../models';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { logViewEvent } from '../../lib/metrics';
+
+interface ButtonDownloadRecoveryKeyProps {
+  email: string;
+  value: string;
+}
+
+export const ButtonDownloadRecoveryKey = ({
+  email,
+  value,
+}: ButtonDownloadRecoveryKeyProps) => {
+  // TODO: format by locale
+  const currentDate = new Date();
+  const downloadDateForFilename = currentDate.toISOString().split('T')[0];
+
+  const downloadDateInLocale = currentDate.toLocaleDateString(
+    navigator.language
+  );
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const fileHeading = ftlMsgResolver.getMsg(
+    'recovery-key-file-header',
+    'SAVE YOUR ACCOUNT RECOVERY KEY'
+  );
+
+  const fileInstructions = ftlMsgResolver.getMsg(
+    'recovery-key-file-instructions',
+    'Store this file containing your account recovery key in a place you’ll remember. Or print it and keep a physical copy. Your account recovery key can help you recover Firefox data if you forget your password.'
+  );
+
+  const fileKey = ftlMsgResolver.getMsg(
+    'recovery-key-file-key-value',
+    `🔑 Key:  ${value}`,
+    { value }
+  );
+
+  const fileUserEmail = ftlMsgResolver.getMsg(
+    'recovery-key-file-user-email',
+    `➤ Firefox account: ${email}`,
+    { email }
+  );
+
+  const fileDate = ftlMsgResolver.getMsg(
+    'recovery-key-file-download-date',
+    `➤ Key generated: ${downloadDateInLocale}`,
+    { date: downloadDateInLocale }
+  );
+
+  const fileSupport = ftlMsgResolver.getMsg(
+    'recovery-key-file-support',
+    '➤ Learn more about your account recovery key: https://mzl.la/3bNrM1I'
+  );
+
+  const fileContent = new Blob(
+    [
+      fileHeading,
+      '\r\n\r\n',
+      fileInstructions,
+      '\r\n\r\n',
+      fileKey,
+      '\r\n\r\n',
+      fileUserEmail,
+      '\r\n',
+      fileDate,
+      '\r\n',
+      fileSupport,
+    ],
+    {
+      type: 'text/plain',
+    }
+  );
+
+  const onClickLogMetrics = () => {
+    logViewEvent(
+      'flow.settings.account-recovery',
+      `recovery-key.download-option`
+    );
+  };
+
+  return (
+    <FtlMsg id="recovery-key-download-button">
+      <a
+        title="Download"
+        href={URL.createObjectURL(fileContent)}
+        download={`Firefox_Account_Recovery_Key_${email}_${downloadDateForFilename}.txt`}
```

Comment:
Q: Could an `${email}` potentially have non-standard characters that might need to be escaped for use as a [default?] filename?

Obligatory grep:
```sh
git grep -n "download=" packages | cat

packages/fxa-content-server/app/scripts/views/mixins/save-options-mixin.js:63:      <a id="download-link" href="${href}" download="${filename}"></a>
packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.tsx:92:        download={`Firefox_Account_Recovery_Key_${email}_${downloadDateForFilename}.txt`}
packages/fxa-settings/src/components/GetDataTrio/index.tsx:113:          download={`${primaryEmail.email} ${contentType}.txt`}
```

Or, maybe not. 🤷 Looks like the GetDataTrio is using `${primaryEmail.email}` directly without escaping or slugifying and has spaces in filenames.

Although this fxa-content-server mixin seems to display unescaped email address, localized "Firefox Backup Authentication Codes", and then does some truncation to make sure a filename is maybe(?) under 200 characters/bytes, so not sure if that is something we're accounting for here or in fxa-settings' GetDataTrio.
https://github.com/mozilla/fxa/blob/29a973e01552cc2b8d799f497c84429443387324/packages/fxa-content-server/app/scripts/views/mixins/recovery-codes-mixin.js#L72-L81

Some quick (and very ignorant) Googling says that Windows/Linux might be limited to 255 bytes, but I'm not sure if a character is 1 byte, or if you can have 2-4 bytes per character if you can somehow sneak emoji or UTF-16 characters into an email address.

Per https://www.ibm.com/docs/en/db2-for-zos/11?topic=unicode-utfs:
> A code unit is the minimal bit combination that can represent a character. Each UTF uses a different code unit size. For example, UTF-8 is based on 8-bit code units. Therefore, each character can be 8 bits (1 byte), 16 bits (2 bytes), 24 bits (3 bytes), or 32 bits (4 bytes). Likewise, UTF-16 is based on 16-bit code units. Therefore, each character can be 16 bits (2 bytes) or 32 bits (4 bytes).

Also https://www.mtu.edu/umc/services/websites/writing/characters-avoid/ (which is hardly a spec, but I'm not finding great answers/tech specs on a quick search; but maybe https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions comes closest).

Reply:
```sh
Firefox_Account_Recovery_Key__2022-12-31.txt # 44 chars (no email)
Firefox_Account_Recovery_Key_f@b.co_2022-12-31.txt # 50 chars (minimal "f@b.co" email)
```
So if we do plan on doing truncation for the formatted filename [like the old mixin code] I guess the 200 byte limit minus the 44 chars for the prefix and suffix + extension still leaves around 156 characters for the email portion of the filename. (with barely, slightly more complex `.substring()` logic since we need to clip the middle part of the filename).

Reply:
Good food for thought here. I've added some initial email truncation, might still modify how that is done.

@bcolsson Should we be localizing the filename, or is English okay for this? I'm not sure if there are best practices.

Reply:
@vpomerleau 

This is a good question. I don't think there's a clear answer. From a user perspective, a localized name for "Account Recovery Key" would obviously be easier to understand. That being said, there's lots of potential technical issues this could introduce, for example OS support for things like double byte characters or  translations that contain characters unsupported by filesystems (e.g. `'`).

For those reasons, I typically see filenames output in Latin characters for Japanese content as an example.

Not to mention the length of the translation in some languages is [considerable](https://pontoon.mozilla.org/uk/terminology/all-resources/?search=key&string=254887).

So my recommendation for the actual filename would be to either leave it in English, or perhaps something simpler like `firefox_key` or `firefox_account_key`. If we still wanted to support a localized filename, I believe there are some ways to create a "localized filename" in certain OS (though I don't know any specifics, only that it *can* be done).

Reply:
@bcolsson If you don't have objections, I think I'll leave the filename in English since, as you mentioned, the translations could get quuuuuite lengthy. 👀 I will ask UX though if we could shorten the name (especially considering that we might change "account recovery" to "data recovery" in the future).

Reply:
If shortening the name to something simpler is possible that's the route I'd recommend, otherwise leaving English as-is is an acceptable alternative.